tests/integration/corrupt-dump.tcl

```diff
@@ -148,20 +158,43 @@ test {corrupt payload: load corrupted rdb with no CRC - #3505} {
     kill_server $srv ;# let valgrind look for issues
 }
 
-test {corrupt payload: load corrupted rdb with empty keys} {
+test {corrupt payload: load corrupted rdb with empty keys - without sanitize} {
+    set server_path [tmpdir "server.rdb-corruption-empty-keys-test"]
+    exec cp tests/assets/corrupt_empty_keys.rdb $server_path
+    start_server [list overrides [list "dir" $server_path "dbfilename" "corrupt_empty_keys.rdb" "sanitize-dump-payload" no]] {
+        r select 0
+        assert_equal [r dbsize] 0
+
+        verify_log_message 0 "*skipping empty key: set*" 0
+        verify_log_message 0 "*skipping empty key: list_quicklist*" 0
+        verify_log_message 0 "*skipping empty key: list_quicklist_empty_ziplist*" 0
+        verify_log_message 0 "*skipping empty key: list_ziplist*" 0
+        verify_log_message 0 "*skipping empty key: hash*" 0
+        verify_log_message 0 "*skipping empty key: hash_ziplist*" 0
+        verify_log_message 0 "*skipping empty key: hash_zipmap*" 0
+        verify_log_message 0 "*skipping empty key: zset*" 0
+        verify_log_message 0 "*skipping empty key: zset_ziplist*" 0
+        verify_log_message 0 "*empty keys skipped: 9*" 0
+    }
+}
```

Comment:
let's run that test inside a loop rather than repeat all that code

Reply:
Silly me.